### DISCUSSION TITLE
https://github.com/micronaut-projects/micronaut-data/pull/2740

### DISCUSSION THREAD

**Discussion by ibmsoft** [micronaut-projects/micronaut-data#2851] [Q&A]
this problem for a long time,there are any plans to resolve this problem

**Comment by radovanradic** [comment]
Not in that PR (because paging remains the issue), maybe in some future work but that will require larger refactoring so not sure when could be done.

**Comment by dstepanov** [comment]
At this moment, there is no easy fix. As a workaround, you can do:
1. Find a page of entities without JOINs
2. Do a second query `findByIdIn(List)` with JOINs
